move BatchPermutationOp to caffe2/operators

wat3rBro opened the pull request:
Summary:
This diff moves BatchPermutationOp to `caffe2/operators`. There're two versions inside fbcode, one from `caffe2/fb/operators` and the other from `caffe2/modules/detectron`. The one from fb is more recent, so I move the fb one and remove the detectron one. As a result, no more fork of custom ops and dynamic link is needed any more, so some files and targets should also be removed (otherwise the op will be registered twice).

Steps:
1: move operators (.h/.cc/.cu) from caffe2/fb/operators to caffe2/operators.
2: remove operators (.h/.cc/.cu) inside caffe2/modules/detectron.
3: remove other duplicated versions, including caffe2/caffe2/share/fb/mask_rcnn/vogue, deeplearning, excluding experimental)
4: remove TARGETS reference from caffe2/caffe2/fb/operators/TARGETS.
5: remove various reference from other places (excluding experimental).

Test Plan:
Original test from caffe2/fb/operators:
buck test mode/dev-nosan caffe2/caffe2:caffe2_test_gpu -- BatchPermutationTest

Detectron:
buck run mode/dev-nosan vision/detection/fair/detectron/detectron/tests:test_batch_permutation_op

buck run mode/dev-nosan mobile-vision/XRayMobileV2/detection_caffe2/tests:test_batch_permutation_op

Differential Revision: D18896100

kostmo commented:
## CircleCI build failures summary
As of commit 93091d0:
* **1/1** broken upstream at merge base 679b20b (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=679b20b1e4b79dd3d62d8a8cb625efeb5a51fe29&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=93091d061efb966e10347d913363f7726e44d693).

### 1 failure *not* recognized by patterns:
| Job | Step |
| --- | --- |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>pytorch_windows_build</sup> | <sup>Build</sup> |


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

wat3rBro committed:
move AliasWithNameOp to caffe2/operators

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/31010

Differential Revision: D18871766

fbshipit-source-id: 5d30c48a3e4ff29861d256949f25b4bd0e3ca99b

wat3rBro committed:
move BatchPermutationOp to caffe2/operators (#31008)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/31008

This diff moves BatchPermutationOp to `caffe2/operators`. There're two versions inside fbcode, one from `caffe2/fb/operators` and the other from `caffe2/modules/detectron`. The one from fb is more recent, so I move the fb one and remove the detectron one. As a result, no more fork of custom ops and dynamic link is needed any more, so some files and targets should also be removed (otherwise the op will be registered twice).

Steps:
1: move operators (.h/.cc/.cu) from caffe2/fb/operators to caffe2/operators.
2: remove operators (.h/.cc/.cu) inside caffe2/modules/detectron.
3: remove other duplicated versions, including caffe2/caffe2/share/fb/mask_rcnn/vogue, deeplearning, excluding experimental)
4: remove TARGETS reference from caffe2/caffe2/fb/operators/TARGETS.
5: remove various reference from other places (excluding experimental).

Test Plan:
Original test from caffe2/fb/operators:
buck test mode/dev-nosan caffe2/caffe2:caffe2_test_gpu -- BatchPermutationTest

Detectron:
buck run mode/dev-nosan vision/detection/fair/detectron/detectron/tests:test_batch_permutation_op

buck run mode/dev-nosan mobile-vision/XRayMobileV2/detection_caffe2/tests:test_batch_permutation_op

Differential Revision: D18896100

fbshipit-source-id: 0575c7f60852a13e37f423e7e546be89333253d0
